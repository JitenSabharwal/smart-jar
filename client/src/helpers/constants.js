export const ADD_CONTAINER_REQUEST = 'ADD_CONTAINER_REQUEST'
export const ADD_CONTAINER_SUCCESS = 'ADD_CONTAINER_SUCCESS'
export const ADD_CONTAINER_FAILURE = 'ADD_CONTAINER_FAILURE'
export const UPDATE_CONTAINER_REQUEST = 'UPDATE_CONTAINER_REQUEST'
export const UPDATE_CONTAINER_SUCCESS = 'UPDATE_CONTAINER_SUCCESS'
export const UPDATE_CONTAINER_FAILURE = 'UPDATE_CONTAINER_FAILURE'
export const DELETE_CONTAINER = 'DELETE_CONTAINER'
export const DELETE_ALL_CONTAINERS = 'DELETE_ALL_CONTAINERS'
export const SELECT_CONTAINER = 'SELECT_CONTAINER'
export const CONTAINER_TABLE_HEADS = [
  { id: 'firstName', numeric: false, disablePadding: true, label: 'FirstName' },
  { id: 'lastName', numeric: false, disablePadding: false, label: 'LastName' },
  { id: 'hobbies', numeric: false, disablePadding: false, label: 'Hobbies' },
  { id: 'birthDate', numeric: false, disablePadding: false, label: 'BirthDate' },
  { id: 'url', numeric: false, disablePadding: false, label: 'Profile' },
]
export const DEFAULT_URL = '/uploads/default.jpeg'
export const MAX_LENGTH = 30
export const NAME_REGEX = /^[a-zA-Z ]{2,30}$/
