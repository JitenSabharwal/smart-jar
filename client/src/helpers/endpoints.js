export const ALL_CONTAINER_ENDPOINT = '/api/v1/containers'
export const ADD_CONTAINER_ENDPOINT = '/api/v1/containers/addContainer'
export const DELETE_CONTAINER_ENDPOINT = '/api/v1/containers/removeContainer'
export const DELETE_ALL_CONTAINER_ENDPOINT = '/api/v1/containers/removeAllContainers'
