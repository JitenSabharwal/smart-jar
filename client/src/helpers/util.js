import axios from 'axios'

/**
 * Function to do a prepare request headers for anything required
 */
export const perpareHeader = () => {
  const id = localStorage.getItem('id_token')
  const config = {
    headers: {
      'Authorization': id ? `Bearer ${id}` : '',
    },
  }
  return config
}

/**
 * Function to do a post Request
 * @param {string} url
 * @param {object} payload
 */
export const postRequest = (url, payload) => {
  const headers = perpareHeader()
  return axios.post(url, payload, headers)
}

/**
 * Function to do a get Request
 * @param {string} url
 * @param {object} payload
 */
export const getRequest = (url, payload) => {
  const headers = perpareHeader()
  return axios.get(url, headers)
}
/**
 * Funtion to filter data out
 * @param {array} data [{id: 'redquired', ...container properties}]
 * @param {*} val
 * @param {string} key If not _id any other prop
 */
export const findUser = (data, val, key = '_id') => {
  return data.filter(u => u[key] === val)
}

/**
 * Functiom to load set an initial state
 */
export const loadInitailState = (url) => {
  return new Promise((resolve, reject) => {
    getRequest(url)
      .then(resp => {
        const {success, data} = resp.data
        if (!success) {
          // If there was a problem, we want to
          return reject(resp)
        } else {
          // If login was successful, set the token in local storage
          const state = {
            container: {
              list: data || [],
            },
          }
          resolve(state)
        }
      })
      .catch(err => {
        reject(new Error('Error: ', err))
      })
  })
}

/**
 * Function returns -1, 0, 1
 *  depending on the match of the
 *  property orderBy
 *  of elements a and b
 * @param {object} a
 * @param {object} b
 * @param {string} orderBy
 */
function desc (a, b, orderBy) {
  if (b[orderBy] < a[orderBy]) {
    return -1
  }
  if (b[orderBy] > a[orderBy]) {
    return 1
  }
  return 0
}
/**
 * Function return a Function to sort elements
 * by orderBy
 * in the given order
 * @param {string} order 'desc/asc'
 * @param {string} orderBy '<string property of the sorting object>'
 */
export const getSorting = (order, orderBy) => {
  return order === 'desc' ? (a, b) => desc(a, b, orderBy) : (a, b) => -desc(a, b, orderBy)
}
