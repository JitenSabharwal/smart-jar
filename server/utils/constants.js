export const ALL_CONTAINER_ENDPOINT = '/api/v1/user'
export const ADD_CONTAINER_ENDPOINT = '/api/v1/user/addUser'
export const UPDATE_CONTAINER_ENDPOINT = '/api/v1/user/updateUser'
export const DELETE_CONTAINER_ENDPOINT = '/api/v1/user/removeUser'
export const DELETE_ALL_CONTAINER_ENDPOINT = '/api/v1/user/removeAllUsers'
