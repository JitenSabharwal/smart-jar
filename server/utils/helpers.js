const log = require('./logger/logger')
/**
 * Parse function to prepare output json
 * @param {object} obj {data: <object>, error:<boolean>, errorMsg: <String>}
 */
const parse = obj => {
  const result = {
    success: true,
  }
  if (obj.error) {
    result.success = false
    result.error = obj.error
  }
  result.data = obj.data
  return result
}

/**
 * Function creates output user object from the given data
 * @param {object} obj {
      itemName: String,
      quantity: String,
      timestamp: Date/String(DD/MM/YYYY),
    }
 */
const outputContainer = (arr) => {
  try {
    const containers = {
      itemName: arr[0].itemName,
      details: arr.map(a => ({quantity: a.quantity, timestamp: a.timestamp})).sort((a, b) => +new Date(b.timestamp) - +new Date(a.timestamp)),
    }
    return containers
  } catch (e) {
    log.error(e)
  }
  return {}
}

module.exports = {
  parse,
  outputContainer,
}
