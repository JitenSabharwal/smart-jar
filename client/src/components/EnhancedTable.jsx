import React from 'react'
import PropTypes from 'prop-types'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TablePagination from '@material-ui/core/TablePagination'
import TableRow from '@material-ui/core/TableRow'
import Paper from '@material-ui/core/Paper'
// import Checkbox from '@material-ui/core/Checkbox'
import EnhancedTableHead from './EnhancedTableHead.jsx'
import EnhancedTableToolbar from './EnhancedTableToolbar.jsx'
// import Avatar from '@material-ui/core/Avatar'

// Helpers
import {getSorting} from '../helpers/util'

/**
 * Component return a table for given set of
 *  rowHeads and data for the rows
 * Comonent is selectable table
 */
class EnhancedTable extends React.Component {
  state = {
    order: 'asc',
    orderBy: this.props.orderBy || '',
    selected: [],
    data: this.props.data || [],
    page: 0,
    rowsPerPage: this.props.rowsPerPage || 10,
  };

  handleRequestSort = (event, property) => {
    // TO handle sorting when the table heads are clicked
    const orderBy = property
    let order = 'desc'

    if (this.state.orderBy === property && this.state.order === 'desc') {
      order = 'asc'
    }

    this.setState({ order, orderBy })
  };

  handleSelectAllClick = (event, checked) => {
    // Handleselect all function
    if (checked) {
      this.setState(state => ({ selected: state.data.map(n => n.id) }))
    } else {
      this.setState({ selected: [] })
    }
    const {length} = this.state.selected
    if (length) {
      this.props.handleSelect(this.state.selected[length - 1])
    }
  };

  handleClick = (event, id) => {
    // TO handle click on the table row
    const { selected } = this.state
    const selectedIndex = selected.indexOf(id)
    let newSelected = []
    if (selectedIndex === -1) {
      newSelected = [id]
    } else if (selectedIndex === 0) {
      newSelected = []
    }
    this.setState({ selected: newSelected })
    // Calling the parent event handler to change the user form data
    if (newSelected.length) {
      this.props.handleSelect(id)
    } else {
      this.props.handleSelect('')
    }
  };

  handleChangePage = (event, page) => {
    this.setState({ page })
  };

  handleChangeRowsPerPage = event => {
    this.setState({ rowsPerPage: event.target.value })
  };

  handleDeleteClick = event => {
    this.handleSelectAllClick()
    this.props.handleDelete(event)
  }

  isSelected = id => this.state.selected.indexOf(id) !== -1;

  render () {
    const { classes, tableName } = this.props
    const { data, order, orderBy, selected, rowsPerPage, page } = this.state
    const emptyRows = rowsPerPage - Math.min(rowsPerPage, data.length - page * rowsPerPage)

    return (
      <Paper className={classes.root}>
        <EnhancedTableToolbar selected={selected} numSelected={selected.length} tableName={tableName} handleDelete={this.handleDeleteClick} />
        <div className={classes.tableWrapper}>
          <Table className={classes.table} aria-labelledby='tableTitle'>
            <EnhancedTableHead
              rows={this.props.rowHeads}
              numSelected={selected.length}
              order={order}
              orderBy={orderBy}
              onSelectAllClick={this.handleSelectAllClick}
              onRequestSort={this.handleRequestSort}
              rowCount={data.length}
            />
            <TableBody>
              {this.props.data
                .sort(getSorting(order, orderBy))
                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                .map(n => {
                  const isSelected = this.isSelected(n._id)
                  if (!n.firstName) {
                    return (<div>No  data</div>)
                  }
                  return (
                    <TableRow
                      hover
                      onClick={event => this.handleClick(event, n._id)}
                      role='checkbox'
                      aria-checked={isSelected}
                      tabIndex={-1}
                      key={n._id}
                      selected={isSelected}
                    >
                      {/* <TableCell padding='checkbox'>
                        <Checkbox checked={isSelected} />
                      </TableCell>
                      <TableCell component='th' scope='row' padding='none' >
                        {n.firstName && n.firstName.length > 30 ? n.firstName.substring(0, 25) + '... ' : n.firstName || ''}
                      </TableCell>
                      <TableCell padding='none'>{n.lastName}</TableCell>
                      <TableCell padding='none'>{n.hobbies}</TableCell>
                      <TableCell padding='none'>{n.birthDate}</TableCell>
                      <TableCell scope='row' padding='none'>
                        <Avatar alt='Remy Sharp' src={n.profilePic} />
                      </TableCell> */}
                    </TableRow>
                  )
                })}
              {emptyRows > 0 && (
                <TableRow style={{ height: 49 * emptyRows }}>
                  <TableCell colSpan={4} />
                </TableRow>
              )}
            </TableBody>
          </Table>
        </div>
        <TablePagination
          component='div'
          count={data.length}
          rowsPerPage={rowsPerPage}
          page={page}
          backIconButtonProps={{
            'aria-label': 'Previous Page',
          }}
          nextIconButtonProps={{
            'aria-label': 'Next Page',
          }}
          onChangePage={this.handleChangePage}
          onChangeRowsPerPage={this.handleChangeRowsPerPage}
        />
      </Paper>
    )
  }
}

EnhancedTable.propTypes = {
  classes: PropTypes.object,
  data: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired,
  orderBy: PropTypes.string.isRequired,
  rowHeads: PropTypes.array.isRequired,
  rowsPerPage: PropTypes.number.isRequired,
  tableName: PropTypes.string,
  handleSelect: PropTypes.func,

}

export default EnhancedTable
