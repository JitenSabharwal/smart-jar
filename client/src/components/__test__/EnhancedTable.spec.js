/* eslint-disable no-undef */
import React from 'react'
import Enzyme, {shallow} from 'enzyme'
import EnhancedTable from '../EnhancedTable.jsx'
import Adapter from 'enzyme-adapter-react-16'
import {CONTAINER_TABLE_HEADS} from '../../helpers/constants'
import UserTableStyle from '../../helpers/ComponentStyles/UserTable.jsx'
const data = []
const handleDelete = () => {
  console.log('DeleteClicked')
}
const orderBy = 'firstName'
const rowsPerPage = 10
Enzyme.configure({adapter: new Adapter()})
describe('EnhancedTable', () => {
  test('render', () => {
    const wrapper = shallow(<EnhancedTable
      data={data}
      handleDelete={handleDelete}
      orderBy={orderBy}
      rowHeads={CONTAINER_TABLE_HEADS}
      rowsPerPage={rowsPerPage}
      classes={UserTableStyle({spacing: 8})}
    />)
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})
