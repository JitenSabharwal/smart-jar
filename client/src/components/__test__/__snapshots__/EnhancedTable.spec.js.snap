// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnhancedTable render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EnhancedTable
    classes={
      Object {
        "root": Object {
          "marginTop": NaN,
          "width": "100%",
        },
        "table": Object {
          "minWidth": 1020,
        },
        "tableWrapper": Object {
          "overflowX": "auto",
        },
      }
    }
    data={Array []}
    handleDelete={[Function]}
    orderBy="firstName"
    rowHeads={
      Array [
        Object {
          "disablePadding": true,
          "id": "firstName",
          "label": "FirstName",
          "numeric": false,
        },
        Object {
          "disablePadding": false,
          "id": "lastName",
          "label": "LastName",
          "numeric": false,
        },
        Object {
          "disablePadding": false,
          "id": "hobbies",
          "label": "Hobbies",
          "numeric": false,
        },
        Object {
          "disablePadding": false,
          "id": "birthDate",
          "label": "BirthDate",
          "numeric": false,
        },
        Object {
          "disablePadding": false,
          "id": "url",
          "label": "Profile",
          "numeric": false,
        },
      ]
    }
    rowsPerPage={10}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(EnhancedTableToolbar)
          handleDelete={[Function]}
          numSelected={0}
          selected={Array []}
        />,
        <div
          className={
            Object {
              "overflowX": "auto",
            }
          }
        >
          <WithStyles(Table)
            aria-labelledby="tableTitle"
            className={
              Object {
                "minWidth": 1020,
              }
            }
          >
            <EnhancedTableHead
              numSelected={0}
              onRequestSort={[Function]}
              onSelectAllClick={[Function]}
              order="asc"
              orderBy="firstName"
              rowCount={0}
              rows={
                Array [
                  Object {
                    "disablePadding": true,
                    "id": "firstName",
                    "label": "FirstName",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "lastName",
                    "label": "LastName",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "hobbies",
                    "label": "Hobbies",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "birthDate",
                    "label": "BirthDate",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "url",
                    "label": "Profile",
                    "numeric": false,
                  },
                ]
              }
            />
            <WithStyles(TableBody)>
              <WithStyles(TableRow)
                style={
                  Object {
                    "height": 490,
                  }
                }
              >
                <WithStyles(TableCell)
                  colSpan={4}
                />
              </WithStyles(TableRow)>
            </WithStyles(TableBody)>
          </WithStyles(Table)>
        </div>,
        <WithStyles(TablePagination)
          backIconButtonProps={
            Object {
              "aria-label": "Previous Page",
            }
          }
          component="div"
          count={0}
          nextIconButtonProps={
            Object {
              "aria-label": "Next Page",
            }
          }
          onChangePage={[Function]}
          onChangeRowsPerPage={[Function]}
          page={0}
          rowsPerPage={10}
        />,
      ],
      "className": Object {
        "marginTop": NaN,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleDelete": [Function],
          "numSelected": 0,
          "selected": Array [],
          "tableName": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(Table)
            aria-labelledby="tableTitle"
            className={
              Object {
                "minWidth": 1020,
              }
            }
          >
            <EnhancedTableHead
              numSelected={0}
              onRequestSort={[Function]}
              onSelectAllClick={[Function]}
              order="asc"
              orderBy="firstName"
              rowCount={0}
              rows={
                Array [
                  Object {
                    "disablePadding": true,
                    "id": "firstName",
                    "label": "FirstName",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "lastName",
                    "label": "LastName",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "hobbies",
                    "label": "Hobbies",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "birthDate",
                    "label": "BirthDate",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "url",
                    "label": "Profile",
                    "numeric": false,
                  },
                ]
              }
            />
            <WithStyles(TableBody)>
              <WithStyles(TableRow)
                style={
                  Object {
                    "height": 490,
                  }
                }
              >
                <WithStyles(TableCell)
                  colSpan={4}
                />
              </WithStyles(TableRow)>
            </WithStyles(TableBody)>
          </WithStyles(Table)>,
          "className": Object {
            "overflowX": "auto",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-labelledby": "tableTitle",
            "children": Array [
              <EnhancedTableHead
                numSelected={0}
                onRequestSort={[Function]}
                onSelectAllClick={[Function]}
                order="asc"
                orderBy="firstName"
                rowCount={0}
                rows={
                  Array [
                    Object {
                      "disablePadding": true,
                      "id": "firstName",
                      "label": "FirstName",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "lastName",
                      "label": "LastName",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "hobbies",
                      "label": "Hobbies",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "birthDate",
                      "label": "BirthDate",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "url",
                      "label": "Profile",
                      "numeric": false,
                    },
                  ]
                }
              />,
              <WithStyles(TableBody)>
                <WithStyles(TableRow)
                  style={
                    Object {
                      "height": 490,
                    }
                  }
                >
                  <WithStyles(TableCell)
                    colSpan={4}
                  />
                </WithStyles(TableRow)>
              </WithStyles(TableBody)>,
            ],
            "className": Object {
              "minWidth": 1020,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "numSelected": 0,
                "onRequestSort": [Function],
                "onSelectAllClick": [Function],
                "order": "asc",
                "orderBy": "firstName",
                "rowCount": 0,
                "rows": Array [
                  Object {
                    "disablePadding": true,
                    "id": "firstName",
                    "label": "FirstName",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "lastName",
                    "label": "LastName",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "hobbies",
                    "label": "Hobbies",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "birthDate",
                    "label": "BirthDate",
                    "numeric": false,
                  },
                  Object {
                    "disablePadding": false,
                    "id": "url",
                    "label": "Profile",
                    "numeric": false,
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  Array [],
                  <WithStyles(TableRow)
                    style={
                      Object {
                        "height": 490,
                      }
                    }
                  >
                    <WithStyles(TableCell)
                      colSpan={4}
                    />
                  </WithStyles(TableRow)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(TableCell)
                      colSpan={4}
                    />,
                    "style": Object {
                      "height": 490,
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "colSpan": 4,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backIconButtonProps": Object {
            "aria-label": "Previous Page",
          },
          "component": "div",
          "count": 0,
          "nextIconButtonProps": Object {
            "aria-label": "Next Page",
          },
          "onChangePage": [Function],
          "onChangeRowsPerPage": [Function],
          "page": 0,
          "rowsPerPage": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(EnhancedTableToolbar)
            handleDelete={[Function]}
            numSelected={0}
            selected={Array []}
          />,
          <div
            className={
              Object {
                "overflowX": "auto",
              }
            }
          >
            <WithStyles(Table)
              aria-labelledby="tableTitle"
              className={
                Object {
                  "minWidth": 1020,
                }
              }
            >
              <EnhancedTableHead
                numSelected={0}
                onRequestSort={[Function]}
                onSelectAllClick={[Function]}
                order="asc"
                orderBy="firstName"
                rowCount={0}
                rows={
                  Array [
                    Object {
                      "disablePadding": true,
                      "id": "firstName",
                      "label": "FirstName",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "lastName",
                      "label": "LastName",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "hobbies",
                      "label": "Hobbies",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "birthDate",
                      "label": "BirthDate",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "url",
                      "label": "Profile",
                      "numeric": false,
                    },
                  ]
                }
              />
              <WithStyles(TableBody)>
                <WithStyles(TableRow)
                  style={
                    Object {
                      "height": 490,
                    }
                  }
                >
                  <WithStyles(TableCell)
                    colSpan={4}
                  />
                </WithStyles(TableRow)>
              </WithStyles(TableBody)>
            </WithStyles(Table)>
          </div>,
          <WithStyles(TablePagination)
            backIconButtonProps={
              Object {
                "aria-label": "Previous Page",
              }
            }
            component="div"
            count={0}
            nextIconButtonProps={
              Object {
                "aria-label": "Next Page",
              }
            }
            onChangePage={[Function]}
            onChangeRowsPerPage={[Function]}
            page={0}
            rowsPerPage={10}
          />,
        ],
        "className": Object {
          "marginTop": NaN,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleDelete": [Function],
            "numSelected": 0,
            "selected": Array [],
            "tableName": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(Table)
              aria-labelledby="tableTitle"
              className={
                Object {
                  "minWidth": 1020,
                }
              }
            >
              <EnhancedTableHead
                numSelected={0}
                onRequestSort={[Function]}
                onSelectAllClick={[Function]}
                order="asc"
                orderBy="firstName"
                rowCount={0}
                rows={
                  Array [
                    Object {
                      "disablePadding": true,
                      "id": "firstName",
                      "label": "FirstName",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "lastName",
                      "label": "LastName",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "hobbies",
                      "label": "Hobbies",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "birthDate",
                      "label": "BirthDate",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "url",
                      "label": "Profile",
                      "numeric": false,
                    },
                  ]
                }
              />
              <WithStyles(TableBody)>
                <WithStyles(TableRow)
                  style={
                    Object {
                      "height": 490,
                    }
                  }
                >
                  <WithStyles(TableCell)
                    colSpan={4}
                  />
                </WithStyles(TableRow)>
              </WithStyles(TableBody)>
            </WithStyles(Table)>,
            "className": Object {
              "overflowX": "auto",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-labelledby": "tableTitle",
              "children": Array [
                <EnhancedTableHead
                  numSelected={0}
                  onRequestSort={[Function]}
                  onSelectAllClick={[Function]}
                  order="asc"
                  orderBy="firstName"
                  rowCount={0}
                  rows={
                    Array [
                      Object {
                        "disablePadding": true,
                        "id": "firstName",
                        "label": "FirstName",
                        "numeric": false,
                      },
                      Object {
                        "disablePadding": false,
                        "id": "lastName",
                        "label": "LastName",
                        "numeric": false,
                      },
                      Object {
                        "disablePadding": false,
                        "id": "hobbies",
                        "label": "Hobbies",
                        "numeric": false,
                      },
                      Object {
                        "disablePadding": false,
                        "id": "birthDate",
                        "label": "BirthDate",
                        "numeric": false,
                      },
                      Object {
                        "disablePadding": false,
                        "id": "url",
                        "label": "Profile",
                        "numeric": false,
                      },
                    ]
                  }
                />,
                <WithStyles(TableBody)>
                  <WithStyles(TableRow)
                    style={
                      Object {
                        "height": 490,
                      }
                    }
                  >
                    <WithStyles(TableCell)
                      colSpan={4}
                    />
                  </WithStyles(TableRow)>
                </WithStyles(TableBody)>,
              ],
              "className": Object {
                "minWidth": 1020,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "numSelected": 0,
                  "onRequestSort": [Function],
                  "onSelectAllClick": [Function],
                  "order": "asc",
                  "orderBy": "firstName",
                  "rowCount": 0,
                  "rows": Array [
                    Object {
                      "disablePadding": true,
                      "id": "firstName",
                      "label": "FirstName",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "lastName",
                      "label": "LastName",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "hobbies",
                      "label": "Hobbies",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "birthDate",
                      "label": "BirthDate",
                      "numeric": false,
                    },
                    Object {
                      "disablePadding": false,
                      "id": "url",
                      "label": "Profile",
                      "numeric": false,
                    },
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    Array [],
                    <WithStyles(TableRow)
                      style={
                        Object {
                          "height": 490,
                        }
                      }
                    >
                      <WithStyles(TableCell)
                        colSpan={4}
                      />
                    </WithStyles(TableRow)>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <WithStyles(TableCell)
                        colSpan={4}
                      />,
                      "style": Object {
                        "height": 490,
                      },
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "colSpan": 4,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backIconButtonProps": Object {
              "aria-label": "Previous Page",
            },
            "component": "div",
            "count": 0,
            "nextIconButtonProps": Object {
              "aria-label": "Next Page",
            },
            "onChangePage": [Function],
            "onChangeRowsPerPage": [Function],
            "page": 0,
            "rowsPerPage": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
